linode/instancedisk/resource.go

```diff
@@ -278,11 +278,20 @@ func deleteResource(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	}
 
 	tflog.Info(ctx, "Deleting instance disk")
-	err = client.DeleteInstanceDisk(ctx, linodeID, id)
+	p, err := client.NewEventPoller(ctx, linodeID, linodego.EntityLinode, linodego.ActionDiskDelete)
```

Comment:
```suggestion
	p, err := client.NewEventPollerWithSecondary(
		ctx,
		linodeID,
		linodego.EntityLinode,
		id,
		linodego.ActionDiskDelete,
	)
```
Let's also check the secondary entity, which is the disk.